### DISCUSSION TITLE
Announcing CFEngine Build

### DISCUSSION THREAD

**Discussion by cody-valle** [cfengine/core#4846] [General]
ICYMI: Yesterday we officially announced the launch of CFEngine Build, which provides a place for users to find and share reusable modules for CFEngine. You can read more about this announcement [here](https://cfengine.com/blog/2021/cfengine-build-launched/). 

Feel free to share your feedback/ideas on CFEngine Build.

**Comment by zzamboni** [comment]
Congrats! This is great stuff, and finally provides a platform for the community to grow and help each other by sharing modules.

One question: I cannot find a single link to https://build.cfengine.com/ from the main https://cfengine.com/ web site. I think this should be prominently featured.

**Reply by nickanderson** [reply]
Thanks @zzamboni.

Indeed, we need to get some content in the main site, I'm sure it's on its way.

@craigcomstock and I are going to chat about this Wednesday in the next Agent is In. https://cfengine.com/webinars/

Would be great to see you there if you can make it.

**Reply by zzamboni** [reply]
Thanks! I've been meaning to make it to those since the beginning, but the timing is not great at the moment. Hopefully I'll get a chance one of these days! :)

**Reply by olehermanse** [reply]
@zzamboni Thanks for the feedback, it will be featured on the main site, it's on the way 👍

**Reply by olehermanse** [reply]
@zzamboni I just wanted to notify you that this was added a while back, there are now a few different links to CFEngine Build from cfengine.com :)

**Comment by basvandervlies** [comment]
Congrat wih the `cfbs` system . Just played with it and created some different masterfiles setup.  It is really easy to build a setup. @nickanderson has done some initial setup for the `scl` library (Thanks). 

I have a question. The SCL framework is heavily depended on mustache/json files the copy is done with  the cf-serverd shortcut `templates`.  The default value is `$(sys.workdir)/templates`. 

How can I install/update these files in the `cfbs`?  Must I add some documentation how to copy/update these files?

**Reply by olehermanse** [reply]
@basvandervlies Can you give some examples of these templates? Are they edited by you (the library author) or users of the library?

**Reply by nickanderson** [reply]
Right, I noticed this when I played with making the `scl` library a module. 
All I did was lay down this `cfbs.json` in the root of my fork of the repo: https://github.com/nickanderson/cf_surfsara_lib/blob/58e4f8bd0a75928abb0461f8a10f68fdbd1c052b/cfbs.json

That makes it work to "add" the repository directly, e.g. `cfbs add  https://github.com/nickanderson/cf_surfsara_lib`

The templates (https://github.com/nickanderson/cf_surfsara_lib/tree/master/templates) are expected to be found as @basvandervlies  notes, in `$(sys.workdir)/templates` which is outside of masterfiles, and thus outside `cfbs` management. `cfbs` only works to put together the policy set.

So, you would have to separately install templates.

Alternatively, templates can be shipped inside of the built policy and there can be a policy to make sure they are in place at `$(sys.workdir)/templates`

**Reply by nickanderson** [reply]
@basvandervlies this might do it

https://github.com/nickanderson/cf_surfsara_lib/commit/140bea34c7b275dc8cc1363f4b761abce6b5651d

**Comment by basvandervlies** [comment]
> @basvandervlies this might do it
> 
> [nickanderson/cf_surfsara_lib@140bea3](https://github.com/nickanderson/cf_surfsara_lib/commit/140bea34c7b275dc8cc1363f4b761abce6b5651d)

Thanks @nickanderson for the suggestion did not think about the solution in the autorun setup ;-) At our site we copy all the files in /var/cfengine/masterfiles to inputs with purge options set. Maybe others also use this option then all mustache json files will be copied to all the hosts.

People can add json files to alter their setup for the service but we can make it work to only copy the mustache files and the default.json file. I will look into it and maybe I find a solution that run only on a HUB/policy server.

**Reply by nickanderson** [reply]
Indeed, with that change the templates are shipped to all hosts as part of the policy set (and each host copies from their own inputs to workdir/templates.

Another alternative would be to add a policy to clone your repo to the hub and copy from there to workdir so that you aren't shipping the templates to each host, but then your hub would have to have internet access to clone (which probably isnt an issue).
